### PR TITLE
Email editor inline link tool

### PR DESCRIPTION
**Pull request by ziggabyte**
## Description
This PR adds an inline tool to put links in text blocks. 

## Screenshots
bild

## Notes to reviewer
I added a shortcut, just randomly picked CTRL+Q/CMD+Q because Q seemed to not interfere with any other existing shortcuts, but please suggest other ones if this isn't good!

## Changes
* Adds `linkToolFactory()` function, to hold and create the link tool. 
* Adds `TextBlockListItem` to display errors in text block in the settings sidebar
* Adds logic and tests to the `blockHasErrors` function that checks for errors in text blocks.

## Related issues
none

### CODE REVIEW

**Review by richardolsson — APPROVED** [review]
This looks all good to me!

File: src/features/emails/components/EmailEditor/tools/InlineLink/index.tsx [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        a: {
+          class: true,
+          href: true,
+          target: '_blank',
+        },
+      };
+    }
+
+    get shortcut() {
+      return 'CMD+Q';
```
Comment (henrycatalinismith) [inline src/features/emails/components/EmailEditor/tools/InlineLink/index.tsx:176]: Isn't ⌘Q the keyboard shortcut to close apps in macOS?

Comment (richardolsson) [reply src/features/emails/components/EmailEditor/tools/InlineLink/index.tsx:176]: Yes it is! I thought we had changed this to CMD+K, thank you so much for identifying it. Will you fix this @ziggabyte?

Comment (ziggabyte) [reply src/features/emails/components/EmailEditor/tools/InlineLink/index.tsx:176]: ooops i had it in my notes and overlooked it!! will fix and change to K!

**Pull request merged**